code/parse/parselo.cpp

```diff
@@ -1391,17 +1394,67 @@ void stuff_string_line(SCP_string &outstr)
 // the default string length if using the F_NAME case.
 char *stuff_and_malloc_string(int type, const char *terminators)
 {
-	SCP_string tmp_result;
-
+	SCP_string tmp_result;  
 	stuff_string(tmp_result, type, terminators);
-	drop_white_space(tmp_result);
-
+	tmp_result=message_translate_tokens(tmp_result.c_str()); 
+   // Deal_with_Delimiters(tmp_result); 
+	drop_white_space(tmp_result); 
 	if (tmp_result.empty())
 		return NULL;
-
 	return vm_strdup(tmp_result.c_str());
 }
 
+void Deal_with_Delimiters(SCP_string &str) 
+{   
+
+	std::stack<int> dels;
+	SCP_string ans; 
+	for (int i = 0; i < str.size(); i++) {
+		// If opening delimiter
+		// is encountered
+		if (str[i] == '[') {
+			dels.push(i);
+		}
+
+		// If closing delimiter
+		// is encountered
+		else if (str[i] == ']' && !dels.empty()) {
+			int pos = dels.top();
+			dels.pop();
+			// Length of substring
+			int len = i - 1 - pos;
+			// Extract the substring 
+		      ans = str.substr(pos + 1, len);          
+			size_t pos2 = 0;
+			 
+
+		    auto& default_preset = Control_config_presets[0];
+		    auto& default_bindings = default_preset.bindings;
+			CC_preset new_preset;
+
+			new_preset.bindings.clear();
+			new_preset.type = Preset_t::tbl;
+			auto& new_bindings=new_preset.bindings;  
+
+		    std::vector<CCB> diff;                                  // compare current key config and default config;
+			//std::sort(new_bindings.begin(), new_bindings.end());
+			//std::sort(default_bindings.begin(), default_bindings.end());
+		    //std::set_difference(new_bindings.begin(),
+			//	new_bindings.end(),
+			//	default_bindings.begin(),
+			//	default_bindings.end(),
+			//	std::back_inserter(diff));                         // find remaped keys; 
+
+				SCP_string rep = "you've remaped this key";        //replace the remaped key into tips; 
+			
+			while ((pos2 = str.find(ans, pos2)) != SCP_string::npos) {
+				str.replace(pos2, ans.length(), rep);
+				pos2 += rep.length();
+			}
+			}
+		}
+} 
+
```

Comment:
Several notes here:
* our code style generally prefers using lowercase with underscores for function names. This function should be renamed to `deal_with_delimiters` to stay within our guidelines.
* this code has very inconsistent indentation.
* this function is declared, but never actually called. Is it necessary? Then it should be called somewhere. If not, why is it part of this PR?
* this method uses `std::vector` and `std::stack`. While we do not have an `SCP_stack` yet, we do generally use `SCP_` variants for STL containers in order to have the opportunity to more easily swap in our own allocators, should that be necessary. So, the recommendation here would be to introduce an SCP_stack based on the precedent set by SCP_vector and use that. Assuming, of course, that this function actually is necessary.